### PR TITLE
#150807985 Create version 1 of RESTful API

### PR DESCRIPTION
**Pull request by Arthur236**
The key features added include:
* Users can register and login
* Users can create shopping lists
* Users can edit shopping lists
* Users can get a specific shopping list and delete it
* Users can add items to a list
* Users can get a specific item, edit or delete it

### CODE REVIEW

File: instance/config.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+"""
+Module for configuration settings
+"""
+import os
+
+
+class Config(object):
```
Comment (esirK) [inline instance/config.py:7]: Nyc stuff here

Comment (Chris-Maina) [reply instance/config.py:7]: Good work.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+language: python
+python:
+ - "3.6.1"
+# Install dependencies
+install:
+  - pip install -r requirements.txt
+  - pip install coveralls
+# Run tests
+script:
+  nosetests --with-coverage --cover-package=tests && coverage report
```
Comment (machariamarigi) [inline .travis.yml:10]: `--cover-package=tests` will only show you coverage results for the tests folder. Try `--cover-package=app` to see coverage for code inside of the app folder

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+after_success:
+  - coveralls
+services:
+  - postgresql
+before_script:
+  - psql -c 'create database shopping_list_api_test;' -U postgres
+addons:
+  postgresql: "9.6"
```
Comment (kkipyegon) [inline .travis.yml:18]: Add a new line after line 18. Also, do this for the other files that don't have a blank line at the end of the file.

**Comment by kkipyegon** [comment]
Also, add a Base test to avoid calling the method create_app every time. So the other test cases should inherit from the base test

**Pull request merged**